src/registry.ts

```diff
@@ -0,0 +1,59 @@
+import { 
+    isUndefined, 
+    WeakMapCreate,
+    WeakMapSet,
+    WeakMapGet,
+    ErrorCreate,
+} from './shared';
+import {
+    RedObject,
+    RedFunction,
+    BlueFunction,
+    BlueObject,
+    RedProxyTarget,
+    BlueValue,
+    RedValue,
+    DistortionMap,
+    RedProxy,
+    BlueProxy,
+    BlueProxyTarget,
+} from './types';
+
+export class SandboxRegistry {
```

Comment:
this is shared for all windows associated to a sandbox, basically, can be used by multiple brokers